spec/requests/api/provision_requests_spec.rb

```diff
@@ -95,6 +95,149 @@
+    end
+    let(:floating_ip1) { FactoryGirl.create(:floating_ip, :cloud_network_only => true, :ext_management_system => ems) }
+
+    let(:provreq_body) do
+      {
+        "template_fields" => {"guid" => template.guid},
+        "requester"       => {
+          "owner_first_name" => api_config(:user_name),
+          "owner_last_name"  => api_config(:user_name),
+          "owner_email"      => "user@example.com"
+        }
+      }
+    end
+
+    let(:provreq1_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => false,
+          "placement_availability_zone" => az.id,
+          "cloud_network"               => cloud_network1.id,
+          "cloud_subnet"                => cloud_subnet1.id,
+          "security_groups"             => security_group1.id,
+          "floating_ip_address"         => floating_ip1.id
+        }
+      )
+    end
+
+    let(:provreq2_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_availability_zone" => az.id
+        }
+      )
+    end
+
+    let(:provreq3_body) do
+      provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => true,
+          "placement_availability_zone" => az.id
+        }
+      )
+    end
+
+    let(:expected_provreq_attributes) { %w(id options) }
+
+    let(:expected_provreq_hash) do
+      {
+        "userid"         => api_config(:user),
+        "requester_name" => api_config(:user_name),
+        "approval_state" => "pending_approval",
+        "type"           => "MiqProvisionRequest",
+        "request_type"   => "template",
+        "message"        => /Provisioning/i,
+        "status"         => "Ok"
+      }
+    end
+
+    it "supports manual placement" do
+      api_basic_authorize collection_action_identifier(:provision_requests, :create)
+
+      aws_dialog # Create the AWS Provisioning dialog
+      run_post(provision_requests_url, provreq1_body)
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_provreq_attributes)
+      expect_results_to_match_hash("results", [expected_provreq_hash])
+
+      options = response_hash["results"].first["options"]
+      expect(options["placement_auto"]).to eq([false, 0])
+      expect(options["placement_availability_zone"].first).to eq(az.id)
+      expect(options["cloud_network"].first).to eq(cloud_network1.id)
+      expect(options["cloud_subnet"].first).to eq(cloud_subnet1.id)
+      expect(options["security_groups"].first).to eq(security_group1.id)
+      expect(options["floating_ip_address"].first).to eq(floating_ip1.id)
```

Comment:
If you're putting an expectation on the entirety of the hash, you can use `match`, otherwise `include` if you just want to check certain things are there:

```ruby
expect(response_hash["results"].first["options"]).to include("placement_auto" => a_collection_containing_exactly(false, 0), # use this matcher if the order of the array's elements can't be determined
                                                             "placement_availability_zone" => [az.id], 
# etc...
```